### PR TITLE
Translate Quests Dialogue lines

### PR DESCRIPTION
**Pull request by theobeaudenon**
Translate Quests Dialogue lines

French Demo
demo
demo

### CODE REVIEW

File: src/main/java/com/wynntils/modules/chat/managers/ChatManager.java [hunk 1]
```diff
@@ -609,7 +617,7 @@ private static void sendTranslation(Matcher m) {
                 // ignore
             }
             McIf.mc().addScheduledTask(() ->
-                    ChatOverlay.getChat().printChatMessage(new TextComponentString(TranslationManager.TRANSLATED_PREFIX + prefix + translatedMsg + suffix)));
+                    ChatOverlay.getChat().printChatMessage(new TextComponentString(TranslationManager.TRANSLATED_PREFIX + prefix + org.apache.commons.lang3.StringUtils.stripAccents(translatedMsg) + suffix)));
```
Comment (theobeaudenon) [inline src/main/java/com/wynntils/modules/chat/managers/ChatManager.java]: Accents not well formatted so I removed them

**Comment by Incompleteusern** [comment]
It might be better to check if the player is in dialogue before applying the matcher, which seems a little wonky. Other than that, great job for fixing an unfixed bug!

I think the regex could be changed from

`            "(.*:)(.+(?:(?:SHIFT)|(?:SNEAK))+.+)(.+.+)", Pattern.DOTALL);  // Quest Dialogue`

to

`            "(.+:)(.+(?:SHIFT)|(?:SNEAK).+)(.+)", Pattern.DOTALL);  // Quest Dialogue`,

This is not tested though

File: src/main/java/com/wynntils/modules/chat/managers/ChatManager.java [hunk 2]
```diff
@@ -962,13 +970,15 @@ public static boolean processUserMention(ITextComponent in, ITextComponent origi
                     if (dialogueChat == null) {
                         dialogueChat = newMessage;
                     } else {
-                        if (ChatConfig.INSTANCE.addTimestampsToChat) addTimestamp(newMessage); // add timestamps to new lines for consistency
+                        if (ChatConfig.INSTANCE.addTimestampsToChat)
```
Comment (magicus) [inline src/main/java/com/wynntils/modules/chat/managers/ChatManager.java]: Please add braces when you spit single-line if statements. It's a huge source of bugs and misunderstandings. (Applies to all locations where you have done this)

**Comment by theobeaudenon** [comment]
> It might be better to check if the player is in dialogue before applying the matcher, which seems a little wonky. Other than that, great job for fixing an unfixed bug!
> 
> I think the regex could be changed from
> 
> ` "(.*:)(.+(?:(?:SHIFT)|(?:SNEAK))+.+)(.+.+)", Pattern.DOTALL); // Quest Dialogue`
> 
> to
> 
> ` "(.+:)(.+(?:SHIFT)|(?:SNEAK).+)(.+)", Pattern.DOTALL); // Quest Dialogue`,
> 
> This is not tested though

Your regex doesn't fully match the dialogue

File: src/main/java/com/wynntils/modules/core/config/CoreDBConfig.java [hunk 3]
```diff
@@ -30,7 +30,8 @@
 
     @Setting(displayName = "Update Stream", description = "Which update stream should the mod be on?\n\n" +
             "§2Stable: §rThe mod will only update when a new version is released. Stable versions are generally more stable than Cutting Edge builds.\n\n" +
-            "§4Cutting Edge: §rThe mod will update whenever a new build is released. Cutting Edge builds will include features that are not yet in Stable versions and are currently in development but may also be less stable than Stable versions.", upload = false)
+            "§4Cutting Edge: §rThe mod will update whenever a new build is released. Cutting Edge builds will include features that are not yet in Stable versions and are currently in development but may also be less stable than Stable versions.\n\n"+
+            "§7Disable: §rThe mod will not update when a new version is released. ", upload = false)
```
Comment (magicus) [inline src/main/java/com/wynntils/modules/core/config/CoreDBConfig.java]: This does not seem relevant to the current PR? Merge error?

File: src/main/java/com/wynntils/modules/chat/managers/ChatManager.java [hunk 4]
```diff
@@ -609,7 +618,8 @@ private static void sendTranslation(Matcher m) {
                 // ignore
             }
             McIf.mc().addScheduledTask(() ->
-                    ChatOverlay.getChat().printChatMessage(new TextComponentString(TranslationManager.TRANSLATED_PREFIX + prefix + translatedMsg + suffix)));
+                    ChatOverlay.getChat().printChatMessage(new TextComponentString(translatedMsg == null ?  TranslationManager.UNTRANSLATED_PREFIX +  formatted : TranslationManager.TRANSLATED_PREFIX + prefix + org.apache.commons.lang3.StringUtils.stripAccents(translatedMsg) + suffix)));
```
Comment (magicus) [inline src/main/java/com/wynntils/modules/chat/managers/ChatManager.java]: Are you sure you should call StringUtils.stripAccents when printing the translations? I think that is wrong. I suspect that means that the Swedish characters "åäö" will be replaced with "aao", which is definitely wrong, and will render the translation incorrect and difficult to understand.

This is different from stripping accents from the Wynncraft original texts before sending them to translation, which is OK if the translation service has problem handling those.

**Review by magicus — CHANGES_REQUESTED** [review]

File: src/main/java/com/wynntils/modules/core/config/CoreDBConfig.java (hunk 3 above)
Comment (Incompleteusern) [reply src/main/java/com/wynntils/modules/core/config/CoreDBConfig.java]: See https://github.com/Wynntils/Wynntils/pull/315

**Comment by magicus** [comment]
@theobeaudenon Did you give up on this? I think the premise is sound, it just needs some tweaking. Do you want help with getting it polished and integrated?

**Comment by theobeaudenon** [comment]
Hey, with the new update, the regex doesn't work anymore. I'm working on it and I will make a new PR.

File: src/main/java/com/wynntils/modules/chat/managers/ChatManager.java (hunk 4 above)
Comment (theobeaudenon) [reply src/main/java/com/wynntils/modules/chat/managers/ChatManager.java]: I will add an option to activate the stripAccents.

File: src/main/java/com/wynntils/modules/core/config/CoreDBConfig.java (hunk 3 above)
Comment (theobeaudenon) [reply src/main/java/com/wynntils/modules/core/config/CoreDBConfig.java]: I will remove this

File: src/main/java/com/wynntils/modules/chat/managers/ChatManager.java (hunk 2 above)
Comment (theobeaudenon) [reply src/main/java/com/wynntils/modules/chat/managers/ChatManager.java]: not my code xD 
maybe IDE do this

**Comment by theobeaudenon** [comment]
strip Accents option 
javaw_2021-09-30_20-22-44


With Accent removal option :
javaw_2021-09-29_20-44-20

Without Accent removal option :
javaw_2021-09-29_20-43-52

now Working Without npc name   
javaw_2021-09-30_20-20-02

File: src/main/java/com/wynntils/modules/chat/managers/ChatManager.java [hunk 5]
```diff
@@ -611,8 +632,13 @@ private static void sendTranslation(Matcher m) {
             } catch (InterruptedException e) {
                 // ignore
             }
+            if(TranslationConfig.INSTANCE.removeAccents){
```
Comment (magicus) [inline src/main/java/com/wynntils/modules/chat/managers/ChatManager.java]:
```suggestion
            if (TranslationConfig.INSTANCE.removeAccents) {
```

File: src/main/java/com/wynntils/modules/chat/managers/ChatManager.java [hunk 6]
```diff
@@ -965,13 +991,16 @@ public static boolean processUserMention(ITextComponent in, ITextComponent origi
                     if (dialogueChat == null) {
                         dialogueChat = newMessage;
                     } else {
-                        if (ChatConfig.INSTANCE.addTimestampsToChat) addTimestamp(newMessage); // add timestamps to new lines for consistency
+                        if (ChatConfig.INSTANCE.addTimestampsToChat){
```
Comment (magicus) [inline src/main/java/com/wynntils/modules/chat/managers/ChatManager.java]:
```suggestion
                        if (ChatConfig.INSTANCE.addTimestampsToChat) {
```

**Review by magicus — CHANGES_REQUESTED** [review]
Just fix the whitespaces as suggested, and this looks fine to me! Thank you for fixing this.

**Review by magicus — APPROVED** [review]
Looks good now!

**Pull request merged**